libenv/sysinfo.c

```diff
@@ -3335,6 +3335,46 @@ void GetDefVars(EvalContext *ctx)
         "jq --compact-output --monochrome-output --ascii-output --unbuffered --sort-keys",
         CF_DATA_TYPE_STRING, "invocation,source=agent,command_name=jq");
 }
+
+/*****************************************************************************/
+
+void SysOsVersionMajor(EvalContext *ctx)
+{
+    const char *flavor = EvalContextVariableGetSpecialString(ctx,
+                                                             SPECIAL_SCOPE_SYS,
+                                                             "flavor");
+
+    int major = -1;
+    int minor = -1;
+    char strmajor[PRINTSIZE(major)];
+
+    sscanf(flavor, "%*[^0123456789]%d%*[^0123456789]%d", &major, &minor);
+    if (major != -1)
+    {
+        xsnprintf(strmajor, sizeof(strmajor), "%d", major);
```

Comment:
I don't like this number parsing only to create a string from it again. It's not so hard to identify a start and an end of an integer in a string. And you would not have to worry about `PRINTSIZE()`. Just create a copy of `flavor` so that it can be modified, split it in-place and use it. It's not like we need to optimize this, but we want to look like professionals. :wink: